### PR TITLE
feat: add BatchGetItem to appsync-simulator(#5963)

### PR DESCRIPTION
**Pull request by behrmans**
#### Description of changes

- add BatchGetItem to amplify-appsync-simulator (all credit to [tan-t](https://github.com/aws-amplify/amplify-cli/pull/5965))
- Added reference to path `amplify-e2e-core` in `amplify-migration-tests` package due to errors regarding `amplify-e2e-core` not being found.

#### Issue #, if available

Issue [5963](https://github.com/aws-amplify/amplify-cli/issues/5963)
Copied almost all code from PR [5965](https://github.com/aws-amplify/amplify-cli/pull/5965)
Reason I made new PR was because it seemed like a lot of time had passed (9 months) since the original PR had made progress on the failing test. Please let me know if I should have handled this differently and I will correct any issues ASAP

#### Description of how you validated changes

- ran existing unit tests
- ran unit test for added utility function
- done E2E manual tests and checked that the BatchGetItem operation will returns response like documented

#### Checklist

- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by CurtisHughes — DISMISSED** [review]

File: packages/amplify-appsync-simulator/src/__tests__/data-loader/dynamo-db/utils/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { mapTableObject, unmarshall } from '../../../../data-loader/dynamo-db/utils';
+
+describe('mapTableObject', () => {
+  it(`will returns mapped TableName key object`, () => {
```
Comment (behrmans) [inline packages/amplify-appsync-simulator/src/__tests__/data-loader/dynamo-db/utils/index.test.ts:4]:
```suggestion
  it(`will return mapped TableName key object`, () => {
```

**Comment by behrmans** [comment]
> Can you please add E2E tests.

Is there a place where I should put them? I'm trying to figure out where the e2e tests for the other operations have been placed.

**Review by jettary — APPROVED** [review]

**Comment by jettary** [comment]
Hey, any progress on it?

**Comment by giraudvalentin** [comment]
Hi, same question, because i changed my offline appsync plugin to use this because its "active & maintened" but face this problem too.

To bypass the problem, i have builded the package amplify-appsync-simulator with your code, used `npm i && cd package/amplify-simulator && yarn build` and commited the change to a repository

Then, i can use it in the serverless-appsync-simulator as dependancies (in a fork too) 

Finally, i use this fork in my serverless project !

Thank you very much @behrmans  & others :)

**Comment by behrmans** [comment]
Reason for force-push was branch was so far behind and had trouble rebasing. All seems fine now. I still plan on trying to finish this PR hopefully soon

**Comment by danielleadams** [comment]
@behrmans can we move this to draft since its still a work in progress? Or are you ready for review?

**Comment by a-marcel** [comment]
Any news about this PR / draft ?

**Comment by danielleadams** [comment]
@behrmans are you still working on this?

**Comment by danielleadams** [comment]
Closing due to inactivity.

**Pull request closed without merging**